skbio/stats/composition.py

```diff
@@ -1079,9 +1079,8 @@ def func(x):
 
     for i in range(c-1):
         ratio = (log_mat[:, i].T - log_mat[:, i+1:].T).T
-        m, p = np.apply_along_axis(func,
-                                   axis=0,
-                                   arr=ratio)
+        p = np.array([func(ratio[:, i])[1]
+                      for i in range(ratio.shape[1])])
```

Comment:
same here. `for i in ratio.T` should be better. If I am correct, transpose just changes the strides and not costly at all.